### PR TITLE
[kubeflow 1.8] Update kubeflow/katib manifests from v0.16.0-rc.1

### PR DESCRIPTION
**Pull request by DnPlas**
Sync kubeflow/katib tag v0.16.0-rc.1 manifests.

WG leads @andreyvelich @johnugeorge @kubeflow/wg-automl-leads 
WG liaison @nagar-ajay 
cc: @annajung @jbottum

### CODE REVIEW

File: .github/workflows/katib_kind_test.yaml [hunk 1]
```diff
@@ -56,3 +56,21 @@ jobs:
 
         echo "Waiting for the Experiment to become Succeeded..."
         kubectl wait --for=condition=Succeeded experiments.kubeflow.org -n kubeflow-user --all --timeout 300s
+
+    - name: Save relevant debug artifacts
```
Comment (annajung) [inline .github/workflows/katib_kind_test.yaml:60]: this can be removed now?

Comment (DnPlas) [reply .github/workflows/katib_kind_test.yaml:60]: Sure, will do.

**Review by andreyvelich — COMMENTED** [review]
Thank you @DnPlas!
/lgtm
/assign @tenzen-y @johnugeorge

**Comment by annajung** [comment]
cc @kimwnasptd

**Pull request merged**